### PR TITLE
add bindings for action effects

### PR DESCRIPTION
**Pull request by maichmueller**
Took a shot at adding the TODO listed bindings for actions.

~(also fixes the template definitions of 2 functions in action.hpp being in the cpp file instead of hpp)~

### CODE REVIEW

**Comment by drexlerd** [comment]
Can we please keep the function definitions in the source files to keep the headers cleaner and safe some compile time and binary size.

File: include/mimir/search/action.hpp [hunk 1]
```diff
@@ -145,6 +145,19 @@ class StripsActionPrecondition
     bool is_applicable(const FlatBitset& fluent_state_atoms, const FlatBitset& derived_state_atoms, const FlatBitset& static_initial_atoms) const;
 };
 
+template<PredicateCategory P>
```
Comment (drexlerd) [inline include/mimir/search/action.hpp:148]: the definitions below should remain in the source file

**Comment by maichmueller** [comment]
A template function definition has to be known at declaration time, so they have to be in the .hpp.

**Comment by drexlerd** [comment]
The explicit instantiation should ensure that the definitions are available in the object file during linking stage. Our implicit rule is to put the in the source file when the number of different templates instantiations is fixed. There are only two possible valid template arguments, meaning that we only need two explicit instantiations.

**Comment by maichmueller** [comment]
Oh I see, that's of course correct then. Reverted that change!

**Comment by drexlerd** [comment]
Awesome. You are making Markus very happy :) He has been asking for this a while ago! I will let him know that you added it. I will merge this when the tests finish

**Pull request merged**